### ISSUE TITLE
GitHubAuthMethod ReturnedLoginAuthInfo returns null value due to async nature

### ISSUE THREAD

**Issue by leowmjw** [rajanadar/VaultSharp#55]
Trying to access the returned Vault Client Token to be cached so no need to request new tokens until it is expired.  Basic call the get secrets from Vault works fine.

Tried the following code snippet but seems ReturnedLoginAuthInfo is null:
```
 // As per example
 IAuthMethodInfo authMethod = new GitHubAuthMethodInfo(personalAccessToken);
 // var token = authMethod.ReturnedLoginAuthInfo.ClientToken; -->null!
```

After further investigation; looks as though it is due to the async nature (docs should be updated with a clear example):
```
 // As per example
 IAuthMethodInfo authMethod = new GitHubAuthMethodInfo(personalAccessToken);
 ... // do stuff
 ... // get secrets data from Vault
 // now returns a valid result; can be cached
 var token = authMethod.ReturnedLoginAuthInfo.ClientToken; 
```
Should the be ReturnedLoginAuthInfo method be marked as async so I can use await until the data returns?

Do suggest the correct way if my observation above is not using it correctly.  Thanks!

**Comment by leowmjw** [comment]
Clarification from @rajanadar 
```
Every instance of IAuthMethodInfo has a property called ReturnedLoginAuthInfo
Once you do even 1 authenticated vault api call using the library, the vault token will be available in this ReturnedLoginAuthInfo object.
So hold onto the reference of IAuthMethodInfo you use to create the Vault Client.
And by the way, the library is a little smart.. It doesn't "relogin" on every api call. I use a Lazy evaluator to get the Token only once. That token is reused for all successive calls. (till the app domain recylces or you reinitialize the vault client)
So please take that into account before you go and implement a "token caching"/"relogin mechanism" etc.

In short,
When you initialize the Vault Client, nothing really happens. No login is performed.
When the first Vault API call is requested, the login call happens and the token is stored in the library.
For further API calls, NO relogin happens. The previous token is used.
This Auth Info is available in the IAuthMethodInfo object after step 2.
```

Will add a simple PR to document this behavior.

**Comment by rajanadar** [comment]
Closing this since the docs will now reflect this.
